File: app/src/main/java/me/imzack/app/end/injector/component/MyPlansComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.MyPlansPresenterModule;
import me.imzack.app.end.injector.scope.FragmentScope;
import me.imzack.app.end.view.fragment.MyPlansFragment;

import dagger.Component;

@FragmentScope
@Component(dependencies = AppComponent.class, modules = MyPlansPresenterModule.class)
public interface MyPlansComponent {

    void inject(MyPlansFragment myPlansFragment);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/MyPlansComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.MyPlansPresenterModule
import me.imzack.app.end.injector.scope.FragmentScope
import me.imzack.app.end.view.fragment.MyPlansFragment

@FragmentScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(MyPlansPresenterModule::class))
interface MyPlansComponent {

    fun inject(myPlansFragment: MyPlansFragment)
}
